### PR TITLE
Added check to prevent double cron

### PR DESCRIPTION
**Pull request by TheHollidayInn**

### CODE REVIEW

File: website/client-old/js/controllers/notificationCtrl.js [hunk 1]
```diff
@@ -12,13 +14,17 @@ habitrpg.controller('NotificationCtrl',
     });
 
     function runYesterDailies() {
+      if (isRunningYesterdailies ) return;
```
Comment (SabreCat) [inline website/client-old/js/controllers/notificationCtrl.js]: extra space in parenthetical

File: website/client-old/js/controllers/notificationCtrl.js [hunk 2]
```diff
@@ -12,13 +14,17 @@ habitrpg.controller('NotificationCtrl',
...
+
       var userLastCron = moment(User.user.lastCron).local();
       var userDayStart = moment().startOf('day').add({ hours: User.user.preferences.dayStart });
 
       if (!User.user.needsCron) return;
       var dailys = User.user.dailys;
 
-      if (!Boolean(dailys) || dailys.length === 0) return;
+      if (!Boolean(dailys) || dailys.length === 0) return
```
Comment (SabreCat) [inline website/client-old/js/controllers/notificationCtrl.js]: removed the semi, intentional?

**Review by SabreCat — CHANGES_REQUESTED** [review]
Tiny code style things

File: website/client-old/js/controllers/notificationCtrl.js [hunk 3]
```diff
@@ -32,6 +38,7 @@ habitrpg.controller('NotificationCtrl',
 
       if (yesterDailies.length === 0) {
         User.runCron();
```
Comment (paglias) [inline website/client-old/js/controllers/notificationCtrl.js]: in a .then() maybe ? so it wait for the runCron request to finish (note that I've not checked if runCron returns a promise or not)

File: website/client-old/js/controllers/notificationCtrl.js [hunk 4]
```diff
@@ -62,6 +69,7 @@ habitrpg.controller('NotificationCtrl',
 
           $scope.ageDailies = function () {
             User.runCron();
+            isRunningYesterdailies = false;
```
Comment (paglias) [inline website/client-old/js/controllers/notificationCtrl.js]: .then here too? Also what happens if the modal is closed without starting the day? isRunningYesterdailies will never reset: not sure if it's possible though

**Review by paglias — CHANGES_REQUESTED** [review]

File: website/client-old/js/controllers/notificationCtrl.js (hunk 4 above)
Comment (TheHollidayInn) [reply website/client-old/js/controllers/notificationCtrl.js]: I prevented the modal from being closed

**Review by paglias — APPROVED** [review]

**Review by SabreCat — APPROVED** [review]

**Pull request merged**